client/client_test.go

```diff
@@ -29,6 +30,38 @@ import (
 	"time"
 )
 
+func Test_NewClient_ShouldSendAuthorizationHeaderWhenRetrievingApiLinks(t *testing.T) {
```

Comment:
Maybe we can create a table test for this two methods, but I am not sure if it will make it better... because you need to provide a different setup and that normally makes them more confusing.

Reply:
Yeah I think if there were more than 2 cases or scenarios I might consider it unless it becomes less readable. It's a balance!